docs/core/compatibility/index.md

```diff
@@ -183,9 +183,13 @@ Changes in this category modify the public surface area of a type. Most of the c
 
    The return value of a method or the type of a property or field cannot be modified. For example, the signature of a method that returns an <xref:System.Object> cannot be changed to return a <xref:System.String>, or vice versa.
 
-- ❌ **DISALLOWED: Adding a field to a struct that previously had no state**
+- ❌ **DISALLOWED: Adding an instance field to a struct that has no nonpublic fields**
 
-  Definite assignment rules allow the use of uninitialized variables so long as the variable type is a stateless struct. If the struct is made stateful, code could end up with uninitialized data. This is both potentially a source breaking and a binary breaking change.
+  If a struct has only public fields or has no fields at all, callers can declare locals of that struct type without calling the struct's ctor or first initializing the local to _default(T)_, so long as all public fields are set on the struct before first use. Adding any new fields - public or nonpublic - to such a struct is a source breaking change for these callers, as the compiler will now require the additional fields to be initialized.
```

Comment:
Why does it matter whether or not there were fields before? If there were fields, adding a new field is a breaking change, no?

```C#
struct Foo {
   int X;
   int Y;
   int Z; // new
}

static Foo CreateFoo()
{
   return new Foo { X = 1, Y = 2 }; // error: Z is not initialized
}
```

Reply:
Your example is covered by the description because all the existing fields are public (has no nonpublic fields). So it's disallowed. If any of the existing fields were private, then the struct would have previously had to be initialized to `default(T)`, and that would apply to the new field too. I'll make a suggestion to change the wording from "Adding an instance field to a struct that has no nonpublic fields" to "Adding an instance field to a struct that has only public fields or no fields".